src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -400,15 +400,21 @@ static void uct_rdmacm_cm_handle_event_established(struct rdma_cm_event *event)
     uct_rdmacm_cm_ep_server_conn_notify_cb(cep, UCS_OK);
 }
 
+static const char*
+uct_rdmacm_cm_event_status_str(const struct rdma_cm_event *event)
+{
+    return strerror(abs(event->status));
```

Comment:
according to https://github.com/linux-rdma/rdma-core/blob/a1a9ffb1a1cb314723dbdfb605447d7919872449/librdmacm/cma.h#L89, event->status is the value of errno, which can be passed to `strerror()` directly
so maybe no need uct_rdmacm_cm_event_status_str wrapper?

Reply:
according to man `rdma_get_cm_event(3)`, it is:
- either `-errno`
```
RDMA_CM_EVENT_UNREACHABLE
              Generated on the active side to notify the user that the remote server is not reachable or unable to respond to a connection  request.   If  this
              event  is  generated in response to a UD QP resolution request over InfiniBand, the event status field will contain an errno, if negative, or the
              status result carried in the IB CM SIDR REP message.
```
or the transport-specific reject reason
```
RDMA_CM_EVENT_REJECTED
              Indicates that a connection request or response was rejected by the remote end point.  The event status field will contain the transport specific
              reject reason if available.  Under InfiniBand, this is the reject reason carried in the IB CM REJ message.
```
e.g. usage in the rdmacm:
https://github.com/linux-rdma/rdma-core/blob/131d94c148f734f8a0cd0247713ba51aec5855d4/librdmacm/cma.c#L1119

Reply:
yes but we examine event->status, not return value from rdmacm_get_event
```
EVENT DATA
       Communication event details are returned in the rdma_cm_event structure.  This structure is allocated by the rdma_cm and released by the rdma_ack_cm_event routine.
       Details of the rdma_cm_event structure are given below.
...
       status      Returns any asynchronous error information associated with an event.  The status is zero if the operation was successful, otherwise the status value is
                   non-zero  and is either set to an errno or a transport specific value.  For details on transport specific status values, see the event type information
                   below.

```

Reply:
@dmitrygx i think we can remove `abs` according to the man page

Reply:
> @dmitrygx i think we can remove `abs` according to the man page

yes, I removed and implemented it in the same way as RDMACM handle `event->status` in `rdma_get_request()`

Reply:
i see it still there, what am I missing?
`return strerror(abs(event->status));`

Reply:
> i see it still there, what am I missing?

sorry, I was wrong - yes, it still there... I thought that I update it to be `return strerror(-event->status);`
do you think it is better to use `-` instead of `abs` as RDMACM does?

Reply:
Isn't status already a positive errno (at least according to man page)?

Reply:
> Isn't status already a positive errno (at least according to man page)?

it seems that man page is incorrect, it should say (i.e. add `... an errno, if negative, or ...`):
```
The status is zero if the operation was successful, otherwise the status value is non-zero and is either set to an errno, if negative, or a transport specific value.
```
this will be aligned with the real behavior and with `RDMA_CM_EVENT_UNREACHABLE` event description:
```
the event status field will contain an errno, if negative, or the status result carried in the IB CM SIDR REP message.
```

Reply:
ok, i see, so let's return strerror(-event->status) and comment that rdma_cm returns a negative errno

Reply:
> so let's return strerror(-event->status) and comment that rdma_cm returns a negative errno

done